core/src/main/scala/neotypes/Driver.scala

```diff
@@ -78,7 +73,45 @@ final class Driver[F[_]] private[neotypes] (private val driver: NeoDriver) {
     }
   }
 
-  /** Close the resources assigned to the neo4j driver
+  private[this] def createSession(config: SessionConfig)
+                                 (implicit F: Async[F]): F[Session[F]] =
+    F.makeLock.flatMap { lock =>
+      F.delay {
+        Session(F, driver.asyncSession(config))(lock)
+      }
+    }
+
+  /** Acquire a streaming session to the database with the default config.
+    * @note ensures the session is properly closed after its use.
+    *
+    * @param F asynchronous effect type.
+    * @tparam S streaming type dependant on the effect type F.
+    * @return a single element streaming with the [[StreamingSession]].
+    *
+    * @see [[SessionConfig]].
+    */
+  def streamingSession[S[_]](implicit F: Async[F], S: Stream.Aux[S, F]): S[StreamingSession[S, F]] =
+    streamingSession(SessionConfig.defaultConfig)
+
+  /** Acquire a streaming session to the database.
+    * @note ensures the session is properly closed after its use.
+    *
+    * @param config the [[SessionConfig]] used to initialize the session.
+    * @param F asynchronous effect type.
+    * @tparam S streaming type dependant on the effect type F.
+    * @return a single element streaming with the [[StreamingSession]].
+    */
+  def streamingSession[S[_]](config: SessionConfig)
+                            (implicit F: Async[F], S: Stream.Aux[S, F]): S[StreamingSession[S, F]] = {
+    val session = F.makeLock.flatMap { lock =>
+      F.delay {
+        Session(F, S, driver.rxSession(config))(lock)
```

Comment:
Should "Session" be differentiated by calling it something to call out it is reactive, like "RxSession"?

Reply:
So we have two session classes so we need two factories.
We have three options.

+ Overload the previous factory _(`Session.apply`)_ - this is the one I choose.
+ Use two different methods like `Session.async` & `Session.streaming` _(or `Session.rx`)_ - this is the one I used for `Driver.streamingSession` since overloading won't work and being an user-facing API I wanted it to be more clear.
+ Keep using `apply` but in the companion object of `StreamingSession` - actually maybe this one would have been the best option?

I really do not have any strong reasons for one or the other if you and @dimafeng prefer a different one I would be happy to change them.

---

BTW, the same applies to **Transaction**.